api/envoy/config/core/v3/config_source.proto

```diff
@@ -240,3 +252,32 @@ message ConfigSource {
   // turn expect to be delivered.
   ApiVersion resource_api_version = 6 [(validate.rules).enum = {defined_only: true}];
 }
+
```

Comment:
Why this move?

Reply:
`ExtensionConfigSource` was defined in *api/envoy/config/core/v3/extension.proto* and uses `ConfigSource` (which is defined in this file).
Therefore if we import *api/envoy/config/core/v3/extension.proto* (so we can use its `TypedExtensionConfig`) in this file we create a cyclic dependency.

Two alternatives:

1. Create a new message, similar to `TypedExtensionConfig` that will be used only in this file.
2. Move `ExtensionConfigSource` from *api/envoy/config/core/v3/extension.proto* to a new file.

I think the current approach is the better than the 2 alternatives, but LMK if you think otherwise.

Reply:
SGTM, just wanted to make sure this doesn't break API compatibility.